### DISCUSSION TITLE
Export of all collections has stuck after request via API

### DISCUSSION THREAD

**Discussion by unordentlich** [outline/outline#3635] [Self Hosting] [answered]
I have requested the export of all collections via the API using the endpoint `/api/collections.export_all`. Actually everything should have worked fine and I got back that everything worked and I got an id.

Through this I do get more information about the FileOperation via `/api/fileOperations.info`, however the process hangs and does not complete the export. For over a day now, the export process has therefore been loading in the settings of my Outline instance and restarting an export process via the UI also hangs.

**To Reproduce**
1. Request an export of all collections via the [API](https://www.getoutline.com/developers#tag/Collections/paths/~1collections.export_all/post)
2. Check if the export will be finished at `/settings/export`
3. Come back after a while and note, that the export is still loading (even if it should only take like 20 or 30 seconds since it has around 90kb)

**Expected behavior**
The export should be completed correctly and the FileOperation should be marked as "finished" after a short time.

**Screenshots**
![image](https://user-images.githubusercontent.com/56507045/172029284-a6be39c3-cce3-44ce-82fe-a423efd277aa.png)

**Outline (please complete the following information):**
- Install: self hosted
- Version: 2d7dd558a1f5fbaf25d6acdcfbfe69f415823442

**Desktop (please complete the following information):**
 - OS: Windows 10/11
 - Browser: Chrome v101.0.4951.67

**Comment by tommoor** [comment]
Please provide the server logs to debug.

If it works via the UI and not via the API then the problem is in your usage and this should be moved to discussions.

**Comment by unordentlich** [comment, marked as answer]
Oh, sure! My bad, I'm sorry. Here are the required information ✌

My browser console provides this error:
![image](https://user-images.githubusercontent.com/56507045/172045515-de11ea12-8591-47c3-831c-34b2e2d444cd.png)

It worked via the UI, but not since exporting via the API. Any further export requests, whether UI or API, are stuck.

This are the server logs:
<pre>{"label":"http","level":"info","message":"  --> POST /api/collections.export_all 200 49ms 636b"}
/opt/outline/node_modules/redis-parser/lib/parser.js:179
    return new ReplyError(string)
           ^

ReplyError: READONLY You can't write against a read only replica. script: 8f55ae4a3be429c6d38c5d5db3e80edf89197b64, on @user_script:41.
    at parseError (/opt/outline/node_modules/redis-parser/lib/parser.js:179:12)
    at parseType (/opt/outline/node_modules/redis-parser/lib/parser.js:302:14) {
  command: {
    name: 'evalsha',
    args: [
      '8f55ae4a3be429c6d38c5d5db3e80edf89197b64',
      '6',
      'bull:globalEvents:wait',
      'bull:globalEvents:paused',
      'bull:globalEvents:meta-paused',
      'bull:globalEvents:id',
      'bull:globalEvents:delayed',
      'bull:globalEvents:priority',
      'bull:globalEvents:',
      '',
      '__default__',
      '{"id":"8f2d8134-c195-4884-80c8-0321b9de57c5","name":"fileOperations.create","teamId":"e55b706d-5f60-4f56-b9c6-f3201e30aba0","actorId":"e5a33449-b64f-4465-b351-bc60543ccd54","modelId":"d24ef74e-a4d1-4ffa-909c-594adfb8dd98","collectionId":null,"ip":"*censored*","data":{"type":"import"},"updatedAt":"2022-06-05T10:12:40.461Z","createdAt":"2022-06-05T10:12:40.461Z","userId":null,"documentId":null}',
      '{"removeOnComplete":true,"removeOnFail":true,"attempts":5,"backoff":{"type":"exponential","delay":1000},"delay":0,"timestamp":1654423960473}',
      '1654423960473',
      '0',
      '0',
      '0',
      'LPUSH',
      '4222bdd5-e39a-40a0-b709-8bb694b38f47'
    ]
  }
}</pre>

Thanks in advance! :)

**Reply by tommoor** [reply]
I mean, the error is right there – your Redis instance is read-only for some reason? This doesn't seem to have anything to do with importing or Outline to be honest so I've moved it to a discussion.

**Reply by unordentlich** [reply]
Alright, I'm sorry, thank you! I'm still new to docker as well as redis, my bad! ✌🏻 
Still have a nice day! :)
